File: Pythonleetcode/Leetcode28.py
def strStr( haystack, needle):
    if  needle=="":
        return 0

    if haystack==needle:
            return 0
    for i in range(0,len(haystack)-len(needle)+1):
        print(haystack[i:i+len(needle)])
        if haystack[i:i+len(needle)]==needle:
            return i
    return -1





print(strStr("hell","ll"))

File: CppPractice/Project1/Project1/Leetcode28.cpp
#include <string>

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

#include <algorithm>

#include <map>

#include <bitset>

using namespace std;


class Solution {
public:
	int strStr(string haystack, string needle) {
		if (needle.size() == 0)
		{
			return 0;
		}
		if (needle.size() > haystack.size())
		{
			return -1;
		}
		int ans = 0;

		for (int i = 0; i < haystack.size() - needle.size() + 1; i++)
		{
			for (int j = 0; j < needle.size(); j++)
			{

				if (haystack[i + j] != needle[j])
				{

					break;

				}
				if (j == needle.size() - 1)
				{
					return ans;
				}
			}
			ans++;

		}
		return -1;

	}
};